File: backend/app/api/schemas/health_data.py
from datetime import datetime
from typing import Literal, Optional, List
from pydantic import BaseModel, Field

SourceType = Literal["wearable", "lab", "questionnaire", "manual"]


class HealthDataPointIn(BaseModel):
    metric_key: str = Field(..., description="Canonical metric key, e.g. sleep_duration")
    value: float
    timestamp: datetime
    source: SourceType = "manual"
    unit: Optional[str] = None  # optional; validated/filled by registry


class HealthDataBatchIn(BaseModel):
    user_id: int
    points: List[HealthDataPointIn]


class HealthDataBatchOut(BaseModel):
    inserted: int
    rejected: int
    rejected_reasons: list[str]

File: frontend/src/api/healthData.ts
import axios from "axios";

const API_BASE = "http://localhost:8000/api/v1";

export type HealthDataPointIn = {
  metric_key: string;
  value: number;
  timestamp: string; // ISO
  source: "wearable" | "lab" | "questionnaire" | "manual";
  unit?: string | null;
};

export type HealthDataBatchIn = {
  user_id: number;
  points: HealthDataPointIn[];
};

export async function uploadHealthDataBatch(payload: HealthDataBatchIn) {
  const res = await axios.post(`${API_BASE}/health-data/batch`, payload);
  return res.data as { inserted: number; rejected: number; rejected_reasons: string[] };
}

